syntax/terminal.vim

```diff
@@ -11,7 +11,7 @@ endif
 
 " Simple syntax highlighting for UnicodeTable command {{{2
 syn match TerminalCSIColorCode1 /\[\d\+m/ conceal
-syn match TerminalCSIColorCode2 /\[\d\+;\d\+m/ conceal
+syn match TerminalCSIColorCode2 /\[\d\+\(;\d\+\)\+m/ conceal
```

Comment:
```suggestion
syn match TerminalCSIColorCode /\[[:;0-9]*m/ conceal
```

This would be the only code we'd need for colors. I recommend doing this inside of vim, however, since I'm not sure that my suggestion's `<ESC>` character would be correctly interpreted.